### DISCUSSION TITLE
PEP about multiple dispatch and @typing.overload

### DISCUSSION THREAD

**Discussion by gabrieldemarmiesse** [coady/multimethod#96] [General]
Hi there! I made a very early draft of a PEP that tries to make multiple dispatch work with `@typing.overload` to get type checkers and code editors support. It would be great to have the feedback of the maintainers on this :)

The draft PEP: https://github.com/gabrieldemarmiesse/PEP-draft-multiple-dispatcher
The discussion: https://discuss.python.org/t/multiple-dispatch-based-on-typing-overload/26197/1

**Comment by coady** [comment]
I'll be interested to see the influence of `functools.singledispatch` on the discussion. Namely the different registration styles and namespace lookups.
